language/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -535,62 +534,36 @@ impl DependencyGraph {
-            // dependencies as the existing one.i
-            Entry::Occupied(entry) if entry.get() == &ext_pkg => {
-                let (self_deps, ext_deps) =
-                    pkg_deps_equal(ext_name, &self.package_graph, ext_graph);
-                if self_deps != ext_deps {
-                    bail!(
-                            "Conflicting dependencies found for '{ext_name}' during external resolution by '{resolver}':\n{}{}",
-                            format_deps("\nExternal dependencies not found:", self_deps),
-                            format_deps("\nNew external dependencies:", ext_deps),
-                        );
-                }
-                // check if acyclic to avoid infinite recursion
-                self.check_acyclic()?;
-                self.override_verify(&ext_pkg, ext_name, overrides)?;
-                // same package, no reason to process its dependencies
-                return Ok(());
-            }
-
-            Entry::Occupied(mut entry) => {
-                if let Some(overridden_pkg) = overrides.get(&ext_name) {
-                    // override found - use it and return - no reason to process its dependencies
-                    entry.insert(overridden_pkg.clone());
-                    return Ok(());
-                } else {
-                    bail!(
-                        "Conflicting dependencies found:\n{0} = {1}\n{0} = {2}",
-                        ext_name,
-                        PackageWithResolverTOML(entry.get()),
-                        PackageWithResolverTOML(&ext_pkg),
-                    );
-                }
-            }
-        };
+        if self
+            .process_graph_entry(
+                &ext_pkg,
+                ext_name,
+                overrides,
+                Some(ext_graph),
+                Some(resolver),
+            )?
+            .is_some()
+        {
+            // existing entry was found
+            return Ok(());
+        }
 
         // if we are here, it means that a new package has been inserted into the graph - we need to
         // process its dependencies and add appropriate edges to them
         for dst in ext_graph.neighbors_directed(ext_name, Direction::Outgoing) {
             // unwrap safe as the table must have the package if the graph has it
             let dst_pkg = ext_table.get(&dst).unwrap();
-            // unwrap is safe as all edges have a Dependency weight
-            let ext_dep = ext_graph.edge_weight(ext_name, dst).unwrap();
-            self.package_graph.add_edge(ext_name, dst, ext_dep.clone());
-            self.merge_pgk(
+            self.merge_pkg(
                 dst_pkg.clone(),
                 dst,
                 ext_graph,
                 ext_table,
                 resolver,
                 overrides,
             )?;
+            // unwrap is safe as all edges have a Dependency weight
+            let ext_dep = ext_graph.edge_weight(ext_name, dst).unwrap();
```

Comment:
nit: Is it possible to avoid the `unwrap` by iterating over `edges_directed` instead of `neighbors_direceted`?